File: src/main/java/net/lomeli/trophyslots/client/ClientProxy.java
package net.lomeli.trophyslots.client;

import net.lomeli.trophyslots.core.IProxy;
import net.lomeli.trophyslots.core.capabilities.IPlayerSlots;
import net.lomeli.trophyslots.core.capabilities.PlayerSlotHelper;
import net.minecraft.client.Minecraft;
import net.minecraft.world.entity.player.Player;

public class ClientProxy implements IProxy {
    @Override
    public void updateSlots(int amount) {
        Player player = Minecraft.getInstance().player;
        IPlayerSlots slots = PlayerSlotHelper.getPlayerSlots(player);
        if (slots != null)
            slots.setSlotsUnlocked(amount);
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/client/ClientProxy.kt
package net.lomeli.trophyslots.client

import net.lomeli.trophyslots.TrophySlots
import net.lomeli.trophyslots.client.models.ModelHandler
import net.lomeli.trophyslots.compat.CompatManager
import net.lomeli.trophyslots.core.Proxy

class ClientProxy : Proxy() {
    override fun preInit() {
        super.preInit()
        ModelHandler.registerModels()
    }

    override fun init() {
        super.init()
        registerForgeEvent(EventHandlerClient)
        registerForgeEvent(TrophySlots.modConfig!!)
    }

    override fun postInit() {
        super.postInit()
        CompatManager.initCompatModules()
    }

    override fun resetConfig() {
        TrophySlots.modConfig?.loadConfig()
    }
}